app/actions/websocket/index.ts

```diff
@@ -310,6 +336,49 @@ export async function handleEvent(serverUrl: string, msg: WebSocketMessage) {
         case WebsocketEvents.APPS_FRAMEWORK_REFRESH_BINDINGS:
             break;
 
-        // return dispatch(handleRefreshAppsBindings());
+            // return dispatch(handleRefreshAppsBindings());
+
+        // Calls ws events:
```

Comment:
For my own understanding, you have created a WS Client specific for calls, so why are these events here? Shouldn't they be handled by the 'Calls WS Client' ?

Reply:
Good question -- I never really thought about it, it was designed like this from v1. I would guess that it's the same answer Claudio gave above. And to add to it: the custom ws client is lightweight and lets us tie into the lifecycle events very closely, which we use to control the peer connection (the media tracks and so on). But, yah, we could merge the two in the future, but for now it's not a high priority.

Thanks Avinash!!

Reply:
That's correct, especially on the plugin side where we didn't have a way to access the existing websocket connection and those events had to be registered in a totally separate part of code (during plugin initialization) so we kept a similar structure here since such events had to deal with redux/app state rather than call client (lower level) specific logic (e.g. receiving media tracks).

Reply:
Those events are needed to keep the state consistent (e.g. who's muted/unmuted). Ideally we don't want to miss any of them. When we cannot guarantee that (e.g. a ws reconnect happened) we would likely revert to fetch the whole state through the HTTP API, even though that's not 100% safe either due to possible races. My idea is that at some point it would be best to consider sending the necessary state purely through the websocket channel but this is not implemented yet.

Reply:
I'm made a ticket to investigate that case, to see if we are at least functional after 15 seconds in the bg. https://mattermost.atlassian.net/browse/MM-45972